docs/automated-testing/unit-testing/README.md

```diff
@@ -179,6 +179,16 @@ Some common pitfalls when writing a unit test that are important to avoid:
   the system is not entirely testable. It is best to wrap third party API calls in interfaces or other structures so
   that they do not get invoked in unit tests.
 
+### Integration tests instead of unit tests
+
+In some situations it is worth considering to write integration tests instead of unit tests to provide full test coverage. The prerequisite for this approach to be successful is to have integration being able to execute at a speed comparable to that of unit tests both locally and in a CI environment.
+Most modern web frameworks like ASP.NET or Spring Boot combined with the right mocking or stubbing approach for external dependencies offer excellent capabilities to enable such scenarios for testing.
```

Comment:
I think we are saying "it depends". External dependencies that allow for efficient integration testing might not need to be mocked. An example of this could be an SQL database, instead of mocking the data layer of an application, one could use a local database with some test data in it, leading to a simpler test fixture setup than with mocking. Other more complex dependencies might be best mocked out completely or partially. Generally the mocking strategy needs to be evaluated carefully on a case by case basis to achieve a good trade-off of testing bigger software units efficiently.